### DISCUSSION TITLE
Docs for library and console apps

### DISCUSSION THREAD

**Discussion by WhiteOlivierus** [dotnet/docfx#10452] [Q&A] [answered]
I can't find in the documentation if you can use docfx for library or console apps. Is this possible?

**Comment by KalleOlaviNiemitalo** [comment]
DocFX has features for generating HTML documentation for a C# library.  An API reference that describes the types and members, "conceptual" documentation that does not directly correspond to the API structure, and links between these.

There was some support for documenting libraries in Visual Basic .NET, and even in other programming ecosystems, but I don't know whether any of that works nowadays.

DocFX does not have any features specifically for documenting a console application and its command-line options and process exit codes.  It can be used for that purpose, though.

**Reply by WhiteOlivierus** [reply]
I meanly want to use this for a C# library, could you point me to the documentation that show how to do this? Because I can't seem to find that.

**Reply by KalleOlaviNiemitalo** [reply, marked as answer]
<https://dotnet.github.io/docfx/docs/dotnet-api-docs.html>

See <https://dotnet.github.io/docfx/api/Docfx.html> for an example of the HTML this generates, and <https://github.com/dotnet/docfx/blob/99caca2b069bc3bcd5c8ca4b4aba0506541196bf/docs/docfx.json> for the configuration with which that was generated.

**Reply by WhiteOlivierus** [reply]
Thank you very much for this lead. This seems promising.
